velox/functions/prestosql/ArrayDistinct.cpp

```diff
@@ -140,6 +140,94 @@ class ArrayDistinctFunction : public exec::VectorFunction {
   }
 };
 
+template <>
+class ArrayDistinctFunction<UnknownType> : public exec::VectorFunction {
```

Comment:
Looks like ArrayDistinctFunction<UnknownType> is different from ArrayDistinctFunction<T> only at the implementation of applyFlat. If so, let's just add an applyFlatUnknown method in the original class ArrayDistinctFunction<T> and call applyFlatUnknown when the element type is UNKNOWN.